x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -118,7 +124,20 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
             .total(autoscalingCapacity.total().storage().getBytes() + unassignedBytes + assignedBytes, null)
             .node(maxShardSize, null)
             .build();
-        return new AutoscalingDeciderResult(requiredCapacity, new ReactiveReason(message, unassignedBytes, assignedBytes));
+        return new AutoscalingDeciderResult(
+            requiredCapacity,
+            new ReactiveReason(
+                message,
+                unassignedBytes,
```

Comment:
nit: I wonder if it makes sense to keep the total size and shard ids together as they are related?